server/memphis_handlers_dls_messages.go

```diff
@@ -405,3 +428,39 @@ func GetPoisonedCgsByMessage(station models.Station, messageSeq, partitionNumber
 
 	return poisonedCgs, nil
 }
+
+func (s *Server) sendToDlsStation(station models.Station, messagePayload []byte, headers map[string]string) error {
+	if station.DlsStation != "" {
+		exist, dlsStation, err := db.GetStationByName(station.DlsStation, station.TenantName)
+		if err != nil {
+			return err
+		}
+		if exist {
+			dlsStationName, err := StationNameFromStr(dlsStation.Name)
+			if err != nil {
+				return err
+			}
+			subject := ""
+			shouldRoundRobin := false
+			if dlsStation.Version == 0 {
+				subject = fmt.Sprintf("%s.final", dlsStationName.Intern())
+			} else {
+				shouldRoundRobin = true
+			}
+
+			if shouldRoundRobin {
+				rand.Seed(time.Now().UnixNano())
+				randomIndex := rand.Intn(len(dlsStation.PartitionsList))
+				subject = fmt.Sprintf("%s$%v.final", dlsStationName.Intern(), station.PartitionsList[randomIndex])
+			}
+
+			acc, err := s.LookupAccount(station.TenantName)
+			if err != nil {
+				return err
+			}
+			headers["station"] = station.Name
```

Comment:
add header for the type of the message.
header name: type values: unacked/failed_schema